flocker/ca/_ca.py

```diff
@@ -58,6 +60,15 @@ def __eq__(self, other):
     def __ne__(self, other):
         return not self.__eq__(other)
 
+    @classmethod
+    def generate(cls):
+        """
+        Create a new 4096-bit RSA key pair.
```

Comment:
If we aim to contribute this KeyPair class back to Twisted, perhaps we shouldn't add this Flocker specific factory method to it.

It could just as well be added to a new free function called eg `flocker_key_pair` which returns a `FlockerKeyPair` wrapper with the Flocker standard 4096bit RSA keypair.

You could also break `selfSignedCACertificate` into a standalone function leaving only the comparison improvements in the class.

And in that case, you could even rename `FlockerKeyPair` to something a little more generic eg ComparableKeyPair and then add that as a patch to https://twistedmatrix.com/trac/ticket/7847

I'm not sure whether there's a place in twisted for the self signed certificate helper function, but I have found a twisted branch where I add an example script that creates them....perhaps that could use this function instead.
 * https://twistedmatrix.com/trac/ticket/6744
 * https://github.com/twisted/twisted/compare/trunk...ssl-endpoint-description-docs-6744

I'd forgotten all about it to be honest.